app/assets/javascripts/components/select_items.jsx

```diff
@@ -0,0 +1,50 @@
+class SelectItems extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {value: this.props.items[this.props.user.role_id - 1].id};
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  initHeaders() {
+    const token = document.querySelector("meta[name=csrf-token]").getAttribute('content');
+
+    axios.defaults.headers.common['X-CSRF-Token'] = token
+    axios.defaults.headers.common['Accept'] = 'application/json'
+  }
+
+  handleChange(event) {
+    this.setState({value: event.target.value});	
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.initHeaders();
+    axios.patch('/users/' + this.props.user.id, {user: {
+      role_id: this.state.value
+    }}).then(response => {
+      this.props.updateFunc(response.data);	
+    });
+  }
+
+  render() {
+    const current_user = this.props.current_user;
```

Comment:
to many const, please only use const for constant for general assignment use let